src/Util.h

```diff
@@ -62,7 +64,56 @@ class ScopedTimer {
 	std::chrono::time_point<std::chrono::high_resolution_clock> startTime;
 	std::string name;
 };
-} // namespace util
 
-long getElapsedUsecs(const struct timeval& tp_start, const struct timeval& tp_end);
-long getElapsedUsecs(const struct timeval& tp_start);
+navtypes::points_t transformReadings(const navtypes::points_t& ps,
+									 const navtypes::transform_t& tf);
+navtypes::trajectory_t transformTraj(const navtypes::trajectory_t& traj,
+									 const navtypes::transform_t& tf);
+
+/* Whether the robot in location given by `tf` collides with anything. */
+bool collides(const navtypes::transform_t& tf, const navtypes::obstacles_t& obss);
+bool collides(const navtypes::transform_t& tf, const navtypes::points_t& lms, double radius);
+
+/* Returns the smallest number t in the unit interval [0,1] such that
+ * the point r0 + t(r1-r0) lies on an obstacle boundary. If there is no such point,
+ * returns t=2.0. */
+double obstacleIntersection(const navtypes::point_t& r0, const navtypes::point_t& r1,
+							const navtypes::obstacles_t& obss);
+
+/* Left-multiplying a (map-frame) point_t by this matrix will give the
+ * corresponding point_t in the frame that was rotated counterclockwise
+ * by theta, then shifted by (x,y) along the new (x,y)-axes.
+ *
+ * For example, for (x, y, theta) = (1, 0, pi/2), the origin frame
+ *
+ *   . .
+ *   > .
+ *
+ * becomes
+ *
+ *   ^ .
+ *   . .
+ */
+navtypes::transform_t toTransformRotateFirst(double x, double y, double theta);
+
+// Given a transform, gives the robot pose that would have that transform.
+// That is, the x,y location and angle theta of the robot such that left-multiplying a
+// world-frame landmark location by `trf` will give you the robot-frame landmark location.
+//
+// NOTE: THIS IS AN INVERSE OF toTransform, NOT toTransformRotateFirst.
+// The parameters of toTransformRotateFirst are not a robot pose
+// (rotation and translation operations do not commute).
+navtypes::pose_t toPose(const navtypes::transform_t& trf, double prev_theta);
+double nearestHeadingBranch(double theta, double prev_theta);
+
+navtypes::transform_t toTransform(const navtypes::pose_t& pose);
+
+// One sample from a standard normal distribution.
+// The thread_id is used to choose which random number generator to use.
+// This is important when trying to rerun a particular random seed; each thread
+// needs its own dedicated sequence of random numbers.
+double stdn(int thread_id);
+// Returns the random seed used for the stdn() function
+long getNormalSeed();
```

Comment:
Can we change these to doxygen comments?
```suggestion
/**
   @brief Given a transform, gives the robot pose that would have that transform.
   
   That is, the x,y location and angle theta of the robot such that left-multiplying a
   world-frame landmark location by `trf` will give you the robot-frame landmark location.

   @warning THIS IS AN INVERSE OF toTransform(), NOT toTransformRotateFirst(). 
   The parameters of toTransformRotateFirst() are not a robot pose
   (rotation and translation operations do not commute).
*/
navtypes::pose_t toPose(const navtypes::transform_t& trf, double prev_theta);
double nearestHeadingBranch(double theta, double prev_theta);

navtypes::transform_t toTransform(const navtypes::pose_t& pose);

/**
  @brief Gets one sample from a standard normal distribution.
 
  @param thread_id Used to choose which random number generator to use; 
  this is important when trying to rerun a particular random seed; each thread
  needs its own dedicated sequence of random numbers.
*/
double stdn(int thread_id);
/**
 Returns the random seed used for the stdn() function
*/
long getNormalSeed();
```

Reply:
Good catch, I totally forgot to add docs and clean up the utility functions. I've added documentation to everything, and I've removed a few vestigial methods that we can't use anymore.